### ISSUE TITLE
TemperatureScaling

### ISSUE THREAD

**Issue by Ruhallah93** [EFS-OpenSource/calibration-framework#55]
Why is TemperatureScaling applied to confidences instead of logits?

**Comment by fabiankueppers** [comment]
Due to consistency with the remaining API. The confidence is converted back to logits internally to apply temperature scaling to.

**Comment by Ruhallah93** [comment]
Is the conversion based on the softmax function? And what happens if the activation of the last layer is sigmoid instead of softmax?

**Comment by Ruhallah93** [comment]
I think converting the softmax output to logits is a bad idea because the softmax function is not injective (1-to-1), i.e. see:
https://calvinmccarter.writeas.com/softmax-function
https://math.stackexchange.com/questions/1928820/topology-and-machine-learning-is-the-softmax-a-homeomorphism

On the other hand, there is a hypothesis that the calibrator function should be an alternative to Softmax but it pushes users to use Softmax.
